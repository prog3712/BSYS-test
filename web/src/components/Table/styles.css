table.customTable {
  min-width: 40%;
  background-color: #ffffff;
  border-collapse: collapse;
  border-width: 2px;
  border-color: #979ca8;
  border-style: solid;
  color: #000000;
}

table.customTable td,
table.customTable th {
  border-width: 2px;
  border-color: #979ca8;
  border-style: solid;
  padding: 5px;
}

table.customTable th {
  color: #fff;
}

table.customTable thead {
  background-color: #363638;
}
