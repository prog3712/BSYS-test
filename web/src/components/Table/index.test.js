import { render } from "@testing-library/react";
import Table from "./index";

test("renders the Table component", () => {
  const mockColumns = [
    {
      Header: "Test Header",
      columns: [
        {
          Header: "ID",
          accessor: "id",
        },
        {
          Header: "Test Data",
          accessor: "test",
        },
      ],
    },
  ];

  const mockData = [
    { id: "1", test: "Test 1" },
    { id: "2", test: "Test 2" },
  ];
  const table = render(<Table columns={mockColumns} data={mockData} />);
  expect(table).toMatchSnapshot();
});
