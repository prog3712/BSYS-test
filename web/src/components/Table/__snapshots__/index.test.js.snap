// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Table component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="customTable"
        role="table"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              colspan="2"
              role="columnheader"
            >
              Test Header
            </th>
          </tr>
          <tr
            role="row"
          >
            <th
              colspan="1"
              role="columnheader"
            >
              ID
            </th>
            <th
              colspan="1"
              role="columnheader"
            >
              Test Data
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              1
            </td>
            <td
              role="cell"
            >
              Test 1
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              2
            </td>
            <td
              role="cell"
            >
              Test 2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="customTable"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            colspan="2"
            role="columnheader"
          >
            Test Header
          </th>
        </tr>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
          >
            ID
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Test Data
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            1
          </td>
          <td
            role="cell"
          >
            Test 1
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            2
          </td>
          <td
            role="cell"
          >
            Test 2
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
